### ISSUE TITLE
one-var to allow for no warnings on different types

### ISSUE THREAD

**Issue by spadgos** [eslint/eslint#2448]
Currently, with [`one-var`](http://eslint.org/docs/rules/one-var.html) you can specify "always" and "never" for each of `var`, `let` and `const` but there's no way to specify that you don't care about one particular type.

That is, it'd be helpful to be able to configure this rule as:

```json
{
  "one-var": [2, {
    "const": "whatever"
  }]
}
```

With this configuration, this would be ok:

```js
var a, b;
let c, d;
const e = 1;
const f = 2,
      g = 3;
```

The rationale here is that `const` is quite different to `var` and `let` since you must give them a value at declaration time, so simply 'manually' hoisting the declaration to the top of the scope as you would with `var` or `let` is not equivalent.

Perhaps a real world example:

The style I often use when writing ES5 is:

```js
var func,  // this is the "manual hoisting" I'm talking about
    anotherFunc;

module.exports = {
  // snip the entire module
};
func = function () { ... }
anotherFunc = (function () {
  return function () { .. };
})();
```

I'd like to be able to keep doing this, but using `const`, however with these rules, I'd have to move the definitions of these private functions up to the top of the file, which is ugly and distracting for someone trying to understand the file.

**Comment by nzakas** [comment]
Sounds good. We should just change any undefined options to be undefined rather than defaulting to something.

**Comment by spadgos** [comment]
Cool. Treating undefined options as undefined would be a breaking change though, since currently the behaviour is undefined = "always". Perhaps a different value?

**Comment by nzakas** [comment]
No, I think undefined is correct and expected.
